src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -453,6 +449,18 @@ private bool ShouldAddTransientTransactionError(MongoException exception)
             return false;
         }
 
+        private void UpdateTransactionStateIfRequired(RequestMessage message)
```

Comment:
Remove this method. These lines were in (and should remain in) the `MessageWasProbablySent` method.